<launch>
  <arg name="calib" />
  <arg name="mode" default="1" />
  <arg name="image" default="image" />

  <arg name="namespace" default="dso" />
  <arg name="output" default="screen" />
  <arg name="bag_data" default="true" />

  <node pkg="tf2_ros" type="static_transform_publisher" name="dso_world_to_world"
        args="0.0 0.0 0.0 0.5 -0.5 0.5 -0.5 world dso_world" >
  </node>

  <node pkg="tf2_ros" type="static_transform_publisher" name="camera_world_to_world"
        args="0.0 0.0 0.0 0.5 -0.5 0.5 -0.5 world camera_world" >
  </node>

  <group ns="$(arg namespace)" >

  <node pkg="dso_ros" type="dso_live" name="dso_live"
        args="calib=$(arg calib) mode=$(arg mode)"
        output="$(arg output)">
    <remap from="image" to="$(arg image)" />
  </node>

  </group>

  <!-- Bag the outputs. -->
  <node if="$(arg bag_data)"
        pkg="rosbag" type="record" name="depthest_bagger"
        output="$(arg output)"
        args="/$(arg namespace)/metric/scale
              -o dso">
    <!-- /$(arg namespace)/depth_registered/image_rect     -->
    <!-- /$(arg namespace)/metric/depth_registered/image_rect     -->
  </node>

</launch>
